### PR TITLE
feat: [CDS-98844]: Allow support for applicationRegex in Gitops Sync Step

### PR DESCRIPTION
**Pull request by ashinsabu3**
This will add a new property to gitops sync step. **applicationRegex**
This should be a regex string(validated in code with pattern.compile())

And with this change, only one of either applicationList(existing field) or applicationRegex will be allowed as a property not both. So this will allow either **none of these fields** OR **one of these fields** not both.

1. If neither are there, then the not subschema will pass, and the anyOf passes. (no applicationRegex, no applicationList is valid)
2. If one is there, then the oneOf subschema will pass, and the anyOf passes. (one of them is also valid)
3. If both are there, then
both of the oneOf subschemas will pass, so the oneOf fails
the not subschema will fail
so the anyOf fails (both of them is not valid)

### CODE REVIEW

**Review by abosii-harness — APPROVED** [review]

**Review by mani5h-harness — APPROVED** [review]

File: v0/pipeline.json [hunk 1]
```diff
@@ -44187,6 +44187,9 @@
                     "minLength" : 1
                   } ]
                 },
+                "applicationRegex" : {
+                  "type" : "string"
```
Comment (mani5h-harness) [inline v0/pipeline.json:44191]: do we need minLength here?
If we default to empty string here in UI, the following condition for one of "applicationsList" or "applicationRegex" to be required will always be true

File: v0/pipeline.json [hunk 2]
```diff
@@ -44314,7 +44325,18 @@
...
+            },
+            "anyOf" : [ {
+              "oneOf" : [ {
+                "required" : [ "applicationRegex" ]
+              }, {
+                "required" : [ "applicationsList" ]
+              } ]
+            }, {
+              "not" : {
+                "required" : [ "applicationRegex", "applicationsList" ]
```
Comment (mani5h-harness) [inline v0/pipeline.json:44337]: what does the step do if both are absent?

File: v0/pipeline.json [hunk 3]
```diff
@@ -44240,7 +44243,12 @@
                     "type" : "string"
                   } ]
                 }
-              }
+              },
+              "oneOf" : [ {
```
Comment (mani5h-harness) [inline v0/pipeline.json:44247]: do we need to update this to the below logic too?

File: v1/pipeline/steps/cd/sync-step-info.yaml [hunk 4]
```diff
@@ -49,6 +51,11 @@ allOf:
       oneOf:
         - type: boolean
         - type: string
+  oneOf: 
+  - required: 
+    - applicationsList
+  - required:
+    - applicationRegex
```
Comment (mani5h-harness) [inline v1/pipeline/steps/cd/sync-step-info.yaml:58]: do we need the same logic that we have below here?

File: v0/pipeline.json (hunk 2 above)
Comment (ashinsabu3) [reply v0/pipeline.json:44337]: If there is a gitops app from previous step(UpdateGitopsApp, FetchLinkedApps) which is present in sweepingoutput that app is by default appended to list of apps that will be synced

File: v1/pipeline/steps/cd/sync-step-info.yaml (hunk 4 above)
Comment (ashinsabu3) [reply v1/pipeline/steps/cd/sync-step-info.yaml:58]: Yes fixing

File: v0/pipeline.json (hunk 1 above)
Comment (ashinsabu3) [reply v0/pipeline.json:44191]: On UI we should implement with removing field completely of whichever one's radio is not checked @litvinovstas

**Review by mani5h-harness — APPROVED** [review]

**Review by yogesh-chauhan — APPROVED** [review]

**Pull request merged**